File: 1. Creational Design Patterns/Singleton/Go/SingletonTemplate.go
package main

// Singleton Pattern – Go Template
// Thread-safe lazy init using sync.Once

import "sync"

type singleton struct{}

var (
	inst *singleton
	once sync.Once
)

func Instance() *singleton {
	once.Do(func() { inst = &singleton{} })
	return inst
}

func main() { _ = Instance() }

File: 1. Creational Design Patterns/Singleton/Java/SingletonTemplate.java
// Singleton Pattern – Java Template
// Static holder idiom for thread-safe lazy init

class Singleton {
    private Singleton() {}
    private static class Holder { private static final Singleton INSTANCE = new Singleton(); }
    public static Singleton getInstance() { return Holder.INSTANCE; }
}

public class SingletonTemplate {
    public static void main(String[] args) {
        Singleton s = Singleton.getInstance();
    }
}
